lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -78,30 +77,31 @@ sub _save_vars {
     close($fd);
 }
 
+# When changing something here, also take a look at OpenQA::Utils::asset_type_from_setting
 sub detect_asset_keys {
     my ($vars) = @_;
 
     my %res;
+    for my $isokey (qw(ISO), map { "ISO_$_" } (1 .. 9)) {
+        $res{$isokey} = 'iso' if $vars->{$isokey};
+    }
 
-    for my $key (keys(%$vars)) {
-        my $value = $vars->{$key};
-
-        # UEFI_PFLASH_VARS may point to an image uploaded by a previous
-        # test (which we should treat as an hdd asset), or it may point
-        # to an absolute filesystem location of e.g. a template file from
-        # edk2 (which we shouldn't).
-        next if $key eq 'UEFI_PFLASH_VARS' && $value =~ m,^/,;
-
-        # Skip external assets, they're handled elsewhere.
-        next if $vars->{$key . '_URL'} || $vars->{$key . '_DECOMPRESS_URL'};
```

Comment:
Deleting this line only should be a sufficient fix. I though external assets are handled by `create_downloads_list()` in `OpenQA::Utils`.

Reply:
Would be really great if you can cover the regression we hit with an according test extension